doc/website/blog/title-tbd.md

```diff
@@ -0,0 +1,244 @@
+all methods. I'm not a Go expert, so the only way I knew of to do this was with a variadic function:
+```go
+func Service(methods ...Method)
+```
+The problem with this approach is that you can only have one argument type that is variable length. What if we wanted to 
+add additional options to the service annotation later on? This is where I was introduced to the powerful design pattern 
+of [functional options](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis):
+```go
+// ServiceOption configures the entproto.Service annotation.
+type ServiceOption func(svc *service)
+
+// Service annotates an ent.Schema to specify that protobuf service generation is required for it.
+func Service(opts ...ServiceOption) schema.Annotation {
+	s := service{
+		Generate: true,
+	}
+	for _, apply := range opts {
+		apply(&s)
+	}
+	// Default to generating all methods
+	if s.Methods == 0 {
+		s.Methods = MethodAll
+	}
+	return s
+}
+```
+This approach takes in a variable number of functions that are called to set options on a struct, in this case, our 
+service annotation. With this approach, we can implement any number of other options functions aside from `Methods`. 
+Very cool!
+
+### List: The Superior GetAll
+With optional method generation out of the way, we could return our focus to adding `GetAll`. How could we implement 
+this method in a safe fashion? Rotem suggested we base the method off of Google's API Improvment Proposal (AIP) for List, 
+[AIP-132](https://google.aip.dev/132). This approach allows a client to retrieve all entities, but breaks the retrieval 
+up into pages. As an added bonus, it also sounds better than "GetAll"!
+
+
+### List Request
+With this design, a request message would look like:
+```protobuf
+message ListUserRequest {
+  int32 page_size = 1;
+
+  string page_token = 2;
+
+  View view = 3;
+
+  enum View {
+    VIEW_UNSPECIFIED = 0;
+
+    BASIC = 1;
+
+    WITH_EDGE_IDS = 2;
+  }
+}
+```
+
+#### Page Size
+The `page_size` field allows the client to specify the maximum number of entries they want to receive in the 
+response message, subject to a maximum page size of 1000. This eliminates the issue of returning more results than the 
+client can handle in the initial `GetAll` implementation. Additionally, the maximum page size was implemented to prevent 
+a client from overburdening the server.
+
+#### Page Token
+The `page_token` field is a base64-encoded string utilized by the server to determine where the next page begins. An 
+empty token means that we want the first page.
+
+#### View
+The `view` field is used to specify whether the response should return the edge ids associated with the entities.
+
+
+### List Response
+The response message would look like:
+```protobuf
+message ListUserResponse {
+  repeated User user_list = 1;
+
+  string next_page_token = 2;
+}
+```
+
+#### List
+The `*_list` field contains page entities.
```

Comment:
```suggestion
The `user_list` field contains page entities.
```
I think it is okay (and correct) to use `user_list` here. The reader will be smart enough to make the transfer ;-)